tests/Integration/Adapter/Tax/TaxComputerTest.php

```diff
@@ -0,0 +1,166 @@
+        $container = self::$kernel->getContainer();
+        $this->taxComputer = $container->get('prestashop.adapter.tax.tax_computer');
+    }
+
+    /**
+     * @dataProvider getComputePriceData
+     */
+    public function testComputePriceWithTaxes(float $taxRate, int $countryId, array $stateIds, string $priceWithoutTaxes, string $priceWithTaxes, int $behavior): void
+    {
+        $taxRuleGroupId = $this->addTaxRuleGroup($taxRate, $countryId, $stateIds, $behavior);
+        $computedPrice = $this->taxComputer->computePriceWithTaxes(
+            new DecimalNumber($priceWithoutTaxes),
+            new TaxRulesGroupId($taxRuleGroupId),
+            new CountryId($countryId)
+        );
+
+        $this->assertEquals(new DecimalNumber($priceWithTaxes), $computedPrice);
+    }
+
+    /**
+     * @dataProvider getComputePriceData
+     */
+    public function testComputePriceWithoutTaxes(float $taxRate, int $countryId, array $stateIds, string $priceWithoutTaxes, string $priceWithTaxes, int $behavior): void
+    {
+        $taxRuleGroupId = $this->addTaxRuleGroup($taxRate, $countryId, $stateIds, $behavior);
+        $computedPrice = $this->taxComputer->computePriceWithoutTaxes(
+            new DecimalNumber($priceWithTaxes),
+            new TaxRulesGroupId($taxRuleGroupId),
+            new CountryId($countryId)
+        );
+
+        $this->assertEquals(new DecimalNumber($priceWithoutTaxes), $computedPrice);
+    }
+
+    public function getComputePriceData()
+    {
+        $countryId = Country::getByIso('fr');
+        $country = new Country($countryId);
+
+        /* taxes for countries without states */
+        yield [20.0, $country->id, [0],  '10.00', '12.00', 0];
+        yield [6.0, $country->id, [0], '10.00', '10.60', 0];
+        yield [21.3, $country->id, [0], '10.00', '12.13', 0];
+        yield [14.7, $country->id, [0], '8.00', '9.176', 0];
+        // tax rate is rounded to 3.148
+        yield [3.14769, $country->id, [0], '8.666', '8.93880568', 0];
+
+        $countryId = Country::getByIso('us');
+        $country = new Country($countryId);
+        $states = State::getStatesByIdCountry($countryId, true);
+        $state = $states[0];
+        $stateId = (int) $state['id_state'];
+
+        /* taxes for countries with state */
+        yield [20.0, $country->id, [$stateId],  '10.00', '12.00', 0];
+        yield [6.0, $country->id, [$stateId], '10.00', '10.60', 0];
+        yield [21.3, $country->id, [$stateId], '10.00', '12.13', 0];
+        yield [14.7, $country->id, [$stateId], '8.00', '9.176', 0];
+        // tax rate is rounded to 3.148
+        yield [3.14769, $country->id, [$stateId], '8.666', '8.93880568', 0];
+
+        $secondState = $states[1];
+        $secondStateId = (int) $secondState['id_state'];
+
+        /* taxes for countries with states and tax has multiple states and only first one should be taken */
+        yield [20.0, $country->id, [$stateId, $secondStateId],  '10.00', '12.00', 1];
```

Comment:
```suggestion
        yield [20.0, $country->id, [$stateId, $secondStateId],  '10.00', '12.00', TaxCalculator::COMBINE_METHOD];
```

Reply:
We should use the existing const to understand the code here better, and I think one const is missing in TaCalculator we should add `public const ONE_TAX_ONLY_METHOD = 0;`